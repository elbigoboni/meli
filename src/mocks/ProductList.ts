import Product from "../types/Product";

export const products: Product[] = [
  {
    title: "Produto 1",
    description: "lalalalalal",
    price: 2000,
    location: "Lugar do produto",
    image: {
      full: "//placeimg.com/680/680",
      thumb: "//placeimg.com/170/170",
    },
  },
  {
    title: "Produto 1",
    description: "lalalalalal",
    price: 2000,
    location: "Lugar do produto",
    image: {
      full: "//placeimg.com/680/680",
      thumb: "//placeimg.com/170/170",
    },
  },
  {
    title: "Produto 1",
    description: "lalalalalal",
    price: 2000,
    location: "Lugar do produto",
    image: {
      full: "//placeimg.com/680/680",
      thumb: "//placeimg.com/170/170",
    },
  },
  {
    title: "Produto 1",
    description: "lalalalalal",
    price: 2000,
    location: "Lugar do produto",
    image: {
      full: "//placeimg.com/680/680",
      thumb: "//placeimg.com/170/170",
    },
  },
];
