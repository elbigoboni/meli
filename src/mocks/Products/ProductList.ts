import ProductInteface from "../../types/Product";
import ProductDetailInteface from "../../types/ProductDetail";

export const ProductsListMock: ProductInteface[] = [
  {
    author: {
      name: "Author xyz",
    },
    item: {
      id: "ml1762712761",
      title: "macbook",
      categories: ["cat1", "cat2"],
      thumbnail: "string",
      price: {
        amount: 1200,
        currency: "ARS",
      },
      location: "Buenos Aires",
      condition: "new",
      free_shipping: true,
    },
  },
  {
    author: {
      name: "Author xyz2",
    },
    item: {
      id: "ml8718728712",
      title: "ipad",
      categories: ["cat1", "cat2"],
      thumbnail: "string",
      price: {
        amount: 500,
        currency: "ARS",
      },
      location: "San Francisco",
      condition: "new",
      free_shipping: true,
    },
  },
];

export const ProductDetailMock: ProductDetailInteface = {
  author: {
    name: "Author xyz",
  },
  description: "desc xyz",
  item: {
    id: "ml1762712761",
    title: "macbook",
    categories: ["cat1", "cat2"],
    thumbnail: "string",
    price: {
      amount: 1200,
      currency: "ARS",
    },
    location: "Buenos Aires",
    condition: "new",
    free_shipping: true,
  },
  sold_quantity: 12,
};
