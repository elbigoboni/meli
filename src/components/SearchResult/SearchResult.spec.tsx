import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import { SearchResult } from ".";
import { ProductDetailMock } from "../../mocks/Products/ProductList";

describe("<SearchResult />", () => {
  describe("Render", () => {
    test("should render correctly", () => {
      const { container } = render(
        <SearchResult product={ProductDetailMock} />,
        {
          wrapper: MemoryRouter,
        }
      );
      expect(container).toMatchSnapshot();
    });
  });
});
