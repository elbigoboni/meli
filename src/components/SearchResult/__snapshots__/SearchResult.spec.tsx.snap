// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> Render should render correctly 1`] = `
<div>
  <a
    href="/item/ml1762712761"
  >
    <div
      class="sc-gsTCUz bRayvk"
    >
      <div
        class="sc-dlfnbm gPxXLx"
      >
        <img
          alt="Imagem"
          class="sc-hKgILt hBoBKD"
          src="string"
        />
      </div>
      <div
        class="sc-fubCfw kvaFFw"
      >
        <span
          class="sc-iBPRYJ dWHJvy"
        >
          ARSÂ 1,200
          <div
            class="sc-gKsewC kCoKrp"
          />
        </span>
        <h3
          class="sc-eCssSg hMINUV"
        >
          macbook
        </h3>
      </div>
      <span
        class="sc-jSgupP ehmssr"
      >
        Buenos Aires
      </span>
    </div>
    <hr
      class="sc-bdfBwQ iAZiBa"
    />
  </a>
</div>
`;
